src/Mod/Part/Gui/Workbench.cpp

```diff
@@ -46,7 +47,25 @@ using namespace PartGui;
 /// @namespace PartGui @class Workbench
 TYPESYSTEM_SOURCE(PartGui::Workbench, Gui::StdWorkbench)
 
-Workbench::Workbench() = default;
+Workbench::Workbench() {
+    /** If we are to have Sketcher_NewSketch as command in toolbar and menu,
+     then we must assure SketcherGui has already been loaded.
+     By putting this in a try/except block we avoid creating a dependency
+     on sketcher workbench as the import will silently fail if sketcher wb is not built.
+     Note that BUILD_SKETCHER is a cmake-gui option.
+     **/
+
+    const char* code =

```

Comment:
Why resort to Python here instead of `Workbench Manager::getWorkbench()`?

Reply:
I'll look into that option when I get the chance.  Currently, I'm burning the midnight oil trying to update my build system from ubuntu 20.04 to ubuntu 24.04 and running into some issues with the updater locking up trying to convert firefox over to a snap format.   It seems firefox is only available now as a snap.  

After the 2nd try I decided to uninstall firefox and see what happens.  It's all in a VM and I saved the 20.04 snapshat, so I can always revert back to that if all else fails.  Once I get that squared I'm sure I'll have to solve a number of issues trying to get the build working again.

I know import SketcherGui initializes the commands, so that's why I went with that.  If Sketcher isn't loaded before Part, then that icon won't show up even after Sketcher is loaded.  Not sure if getWorkbench() will initialize it or not.  If it does I'll be happy to switch over to that and avoid running the python.

Reply:
It looks like getWorkbench() does not initialize the workbench.  It returns a null pointer if the workbench hasn't been loaded yet.  I don't think that is a bug because it says in the comments it returns 0 (which should probably say null pointer instead).  It works as expected when the workbench has already been initialized, but that's not the case here.

I should also point out I tried, but failed to test this by building FreeCAD without Sketcher workbench.  The icon still appeared on the toolbar and I was still able to create a sketch and edit it even though I unchecked BUILD_SKETCHER checkbox in cmake-gui, clicked Configure, then clicked Generate.  I knew I would also need to uncheck BUILD_PARTDESIGN because that workbench depends on Sketcher module, so I unchecked it.  I also had to uncheck BUILD_ASSEMBLY to get the build to work.  Even unchecking all 3, I still got a working Sketcher and a working PartDesign (didn't look if Assembly was also built or not).  Either I missed some important step (like not cleaning the build directory) or this cmake option to selectively build or not builld certain modules is broken.  I use ccache, so that might also be part of the issue..

My method of testing this PR, then, was to temporarily and intentionally misspell SketcherGui for a test to see what would happen.  As expected, the icon did not appear on the toolbar, and there was no build failure and no exception generated and no error displayed to the user.  A user who has elected not to build Sketcher should not see any errors when opening Part WB.

Let me know if there is another way you would prefer to do this.  I'm not sure why the manipulator code in InitGui.py wasn't working as expected and was putting the icon to the end of the toolbar instead of inserting it before the extrude icon where it was supposed to put it.  That method has the drawback that it gets called every time the workbench is activated, so it's not as efficient as this one, which only gets called during initialization.  As I understand it, the toolbar is simply hidden upon leaving the workbench, and then made visible again upon returning.

It could be better to have another toolbar dedicated to useful icons from other workbenches, maybe PartWBExtras or some such, and it could have this new sketch tool, maybe some draft arrays, draft clone, part design subshapebinder, etc.  This PR lays the foundation for bringing in these other workbench icons without creating the dependencies on them, or at least not a hard dependency.